### PR TITLE
chore(core-react): add typings for button

### PR DESCRIPTION
**Pull request by jgbenito7**
Add typings for Button in core-react.

### CODE REVIEW

File: packages/core-react/index.d.ts [hunk 1]
```diff
@@ -8,4 +8,16 @@ export type TextFieldProps = {
 
 declare const TextField: React.FC<TextFieldProps>;
 
-export { TextField };
+export type ButtonProps = {
```
Comment (jgbenito7) [inline packages/core-react/index.d.ts]: Should we start a `types` folder for the definitions to keep things more organized?

Comment (adamraider) [reply packages/core-react/index.d.ts]: yea

File: packages/core-react/index.d.ts [hunk 2]
```diff
@@ -8,4 +8,16 @@ export type TextFieldProps = {
...
+  Tag?: 'button' | 'a';
+  children?: React.ReactNode;
+  type?: 'primary' | 'secondary' | 'tertiary';
```
Comment (seimithwework) [inline packages/core-react/index.d.ts]: I'm ok with this for now, but I think we should definitely update these to enums in the future.

File: packages/core-react/index.d.ts [hunk 3]
```diff
@@ -8,4 +8,16 @@ export type TextFieldProps = {
...
+  compact?: boolean;
+  danger?: boolean;
+  disabled?: boolean;
+  className?: string;
+};
```
Comment (seimithwework) [inline packages/core-react/index.d.ts]: I think we can add  `& React.HTMLProps<HTMLButtonElement>` at the end here too. (I'll add and push!)

**Pull request merged**